pyteal/ir/tealblock.py

```diff
@@ -204,6 +204,31 @@ def NormalizeBlocks(cls, start: "TealBlock") -> "TealBlock":
                     if prev is start:
                         start = block
 
+        for block in TealBlock.Iterate(start):
+            if len(block.ops) == 0:
+                outgoing = block.getOutgoing()
+                if len(outgoing) == 1:
+                    # if block has 0 ops and 1 outgoing edge, directly connect every incoming block
+                    # to the single outgoing block, thereby removing an unnecessary intermediate
+                    # jump to this block
+                    outgoingBlock = outgoing[0]
+                    for i, incomingBlock in enumerate(outgoingBlock.incoming):
+                        if block is incomingBlock:
+                            # remove block from incoming of outgoing
+                            outgoingBlock.incoming = (
```

Comment:
Is there a difference between using `incoming.remove(i)` here ? Are we afraid of indexing errors?

Reply:
`remove` compares by value (by calling `__eq__`), but here we want to compare by reference. There may be multiple blocks with the same code (e.g. `int 1; return` is a common block that can appear more than once in a program) which return true under value comparison, but false under reference comparison.

Reply:
Sorry, I think I meant `incoming.pop(index)` (or some equivalent of removing an element by the index) -- you're totally right about the `remove` I think.

Reply:
Good idea, pop would be better here -- I'll update this now